giskard/scanner/llm/llm_prompt_injection_detector.py

```diff
@@ -30,88 +27,46 @@ class LLMPromptInjectionDetector(Detector):
     .. [#] Leon Derczynsky, garak:  LLM vulnerability scanner, https://github.com/leondz/garak
     """
 
-    def __init__(self, threshold: float = 0.5):
+    def __init__(self, num_samples: Optional[int] = None, threshold: float = 0.5):
+        self.num_samples = num_samples
         self.threshold = threshold  # default
 
     def get_cost_estimate(self, model: BaseModel, dataset: Dataset) -> float:
+        num_samples = self.num_samples
+        if num_samples is None:
+            data_loader = PromptInjectionDataLoader(num_samples=self.num_samples)
+            num_samples = len(data_loader.df)
```

Comment:
It's not nice that we fire a new download and all the logic to merge and prepare the dataset just to get the number of samples. Could we cache the data loader data in a global variable after preprocessing? Or could we get rid of the whole preprocessing and do it once, exporting the data file and adding it to the giskard repo?

Reply:
done